components/selectors/matching.rs

```diff
@@ -541,6 +545,21 @@ where
 {
     match *selector {
         Component::Combinator(_) => unreachable!(),
+        Component::Slotted(ref selectors) => {
+            context.shared.nesting_level += 1;
```

Comment:
Minor thing, but I wonder if we should have an RAII-like thing to handle the incrementing/decrementing.

Reply:
Yeah, I thought about it to, it's slightly tricky, because it requires grabbing a mutable reference to the context, which is not possible, so we should switch the count to be a `Cell<usize>` instead... Not that it matters a lot, tbh.